helix-core/src/object.rs

```diff
@@ -40,6 +40,48 @@ pub fn select_next_sibling(syntax: &Syntax, text: RopeSlice, selection: Selectio
     })
 }
 
+pub fn select_all_siblings(syntax: &Syntax, text: RopeSlice, selection: Selection) -> Selection {
+    selection.transform_iter(|range| {
+        let mut cursor = syntax.walk();
+        let fallback = || vec![range].into_iter();
+        let (from, to) = range.into_byte_range(text);
+        cursor.reset_to_byte_range(from, to);
+
+        if !cursor.goto_parent_with(|parent| parent.child_count() > 1) {
+            return fallback();
+        }
+
+        select_children(&mut cursor, text, range.direction()).unwrap_or_else(fallback)
+    })
+}
+
+pub fn select_all_children(syntax: &Syntax, text: RopeSlice, selection: Selection) -> Selection {
+    selection.transform_iter(|range| {
+        let mut cursor = syntax.walk();
+        let (from, to) = range.into_byte_range(text);
+        cursor.reset_to_byte_range(from, to);
+        select_children(&mut cursor, text, range.direction())
+            .unwrap_or_else(|| vec![range].into_iter())
+    })
+}
+
+fn select_children<'n>(
+    cursor: &'n mut TreeCursor<'n>,
+    text: RopeSlice,
+    direction: Direction,
+) -> Option<<Vec<Range> as std::iter::IntoIterator>::IntoIter> {
```

Comment:
I would just return a Vec and let the callsite call `into_iter` or if you really want to return an iterator use an anonymous `impl Iterator` instead this is a bit hard to read